### ISSUE TITLE
Dokku installation, UI Disconnected

### ISSUE THREAD

**Issue by hilkoc** [askmike/gekko#1688]
**Note: for support questions, please join our [Discord server](https://discord.gg/26wMygt)**

* **I'm submitting a ...**
[x ] question about installation
[ ] feature request
[ ] bug report

I've spent a lot of time trying to figure it out myself, but got really stuck now. I hope some one can please help with this:

I've got a VPS running ubuntu on which I've installed [dokku](https://github.com/dokku/dokku). Dokku is essentially a bash script that manages a docker container and redeploys the app as soon as you git push to the server.

The deployment almost succeeds, the gui runs and I can see the logs. The only problem is I get the same error as [this issue #1065](https://github.com/askmike/gekko/issues/1065) : Disconnected
Something happened to either Gekko or the connection. Please check the terminal where Gekko is running or your network connection.

From what I understand from reading all other related issues, it has to do with the websocket that cannot connect through nginx to the docker container.

I've spent hours trying different nginx configurations but nothing works. Can you please help?

**Comment by askmike** [comment]
> it has to do with the websocket that cannot connect through nginx to the docker container.

The issue is indeed that your browser cannot open a websocket directly to the docker that runs Gekko. Can you open the devtools of your browser and post a screenshot of the network tab?

**Comment by hilkoc** [comment]
Hi Mike, Thanks for your prompt reply.
After some more tinkering I finally figured it out.

Just in case it might help someone who lands here after a search, here's what I did to get it working.

Install Dokku and the letsencrypt plugin following their respective installation instructions.
Create an app called gekko:  `dokku apps:create gekko`
Clone the code for gekko, (or reset it back to origin/develop after trying out lots of things).
Change the configuration exactly as described in the gekko installation instructions.
Deploy the app by pushing it to your dokku repo.

Now you'll see that it's almost working, except for the UI Disconnected error. To fix this make sure that the proxy mappings for the container are as follows:

```
$>sudo dokku proxy:ports gekko
-----> Port mappings for gekko
-----> scheme             host port         container port                           
http                      3000              3000                                               
https                     3000              3000                                               
https                     443               3000

```

The crucial part here is the middle line. The first and last are already preconfigured by dokku. Because the connection is encrypted we need to add mapping for https 3000:3000 ourselves.

Then it works.
